src/client/datascience/interactive-common/interactiveBase.ts

```diff
@@ -329,6 +331,10 @@ export abstract class InteractiveBase extends WebviewPanelHost<IInteractiveWindo
                 this.handleMessage(message, payload, this.showDataViewer);
                 break;
 
+            case InteractiveWindowMessages.ShowDataWrangler:
+                this.handleMessage(message, payload, this.showDataWrangler);
+                break;
+
             case InteractiveWindowMessages.GetVariablesRequest:
                 this.handleMessage(message, payload, this.requestVariables);
                 break;
```

Comment:
I was thinking the requestTmLanguage here would be in the base class. So you don't need to handle the message here.

Reply:
the ```super.onMessage``` would handle it instead.

Reply:
WebviewPanelHost or WebViewHost would be better?
WebviewPanelHost extends WebViewHost
WebViewHost also has 4 other onMessages already: SharedMessages.Started, CssMessages.GetCssRequest, CssMessages.GetMonacoThemeRequest, and InteractiveWindowMessages.GetHTMLByIdResponse

Leaning towards WebViewHost but I don't have enough knowledge about it

Reply:
I think the lower you can push this stuff the better.